addons/resource_distribution_controller.go

```diff
@@ -0,0 +1,161 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"reflect"
+	"slices"
+
+	templatev1 "github.com/openshift/api/template/v1"
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v4/v1alpha1"
+	"github.com/red-hat-storage/odf-multicluster-orchestrator/controllers/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// ResourceDistributionReconciler reconciles a ManifestWork object
+type ResourceDistributionReconciler struct {
+	Scheme           *runtime.Scheme
+	HubClient        client.Client
+	SpokeClient      client.Client
+	SpokeClusterName string
+	Logger           *slog.Logger
+	CurrentNamespace string
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ResourceDistributionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Logger.Info("Setting up controller with manager")
+	labelSelectorPredicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			utils.CreatedByLabelKey: "odf-multicluster-orchestrator",
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("resource_distribution_controller").
```

Comment:
In the logs the controller name will appear as `resource_distribution_controller` how about `ResourceDistributionController` to make it similar to what we have for other controllers

Reply:
No. It will appear as `ResourceDistributionController` because it's set up that way in the manager.